Z/ZipFlow/build_tarballs.jl

```diff
@@ -0,0 +1,22 @@
+using BinaryBuilder
+
+name = "ZipFlow"
+version = v"1.0.0"
+
+sources = [GitSource("https://github.com/madler/zipflow.git",
+                     "d4d73304252504bbade9aa6f34332bbb00de6664")]
+
+script = raw"""
+    cd ${WORKSPACE}/srcdir/zipflow/
+    cc zipflow.c -shared -fPIC -std=c99 -o ${libdir}/libzipflow.${dlext} -I${includedir} -L${libdir} -lz
```

Comment:
The code uses `unix` as an identifier so I added `-std=c99` to fix that but apparently `-std=c99` means `strdup` ends up undefined when compiling with GCC (but works with Clang). 🙃

Reply:
Let's try this
```suggestion
    cc zipflow.c -shared -fPIC -std=gnu99 -o ${libdir}/libzipflow.${dlext} -I${includedir} -L${libdir} -lz
```

Reply:
They include `sys/stat.h`, but judging by
```console
sandbox:${WORKSPACE} # grep -B1 'define S_IFMT' /opt/${target}/${target}/sys-root/usr/include/sys/stat.h
#if defined __USE_BSD || defined __USE_MISC || defined __USE_XOPEN
# define S_IFMT         __S_IFMT
```
the macro is defined conditionally.  Maybe define `__USE_XOPEN`?

Reply:
Nevermind, that's solved automatically by `gnu99`:
```console
sandbox:${WORKSPACE} # echo '#include <sys/stat.h>' | cc -std=c99 -x c - -dM -E | grep '\<S_IFMT\>'
sandbox:${WORKSPACE} # echo '#include <sys/stat.h>' | cc -std=gnu99 -x c - -dM -E | grep '\<S_IFMT\>'
#define S_IFMT __S_IFMT
```
I'm lagging behind you

Reply:
`gnu99` reintroduces the `unix` macro though, which re-breaks a bunch of stuff 😑

Reply:
Oh, now I understand it.  This is rather...uhm...hilarious.  Well, maybe defining `__USE_XOPEN` without `gnu99` would help?

Reply:
Looks like that leaves `strdup` undefined. Software was a mistake.

Reply:
```console
sandbox:${WORKSPACE} # grep -B1 -r 'define \<strdup\>' /opt/${target}/${target}/sys-root/usr/include/
/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/usr/include/bits/string2.h-#  if defined __USE_SVID || defined __USE_BSD || defined __USE_XOPEN_EXTENDED
/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/usr/include/bits/string2.h:#   define strdup(s) __strdup (s)
```
Try `__USE_BSD`? :grimacing: